test/vtcomposite-param-validation.test.js

```diff
@@ -513,136 +513,130 @@ test('[composite] failure: buffer size is not positive int32', assert => {
-    assert.end();
+    assert.equal(err.message, 'params.language must be null or a string', 'expected error message');
   });
-});
 
-test('[localize] failure: options.compress is not a boolean', assert => {
-  localize(Buffer.from('hello world'), 'en', 'AD', { compress: 1 }, function (err, result) {
+  localize({
+    buffer: Buffer.from('hi'),
+    language: '' // empty string
+  }, function (err) {
     assert.ok(err);
-    assert.ok(/'compress' must be a boolean/.test(err.message), 'expected error message');
-    assert.end();
+    assert.equal(err.message, 'params.language cannot be an empty string', 'expected error message');
   });
-});
 
-test('[localize] success: options may be omitted', assert => {
-  localize(mvtFixtures.get('002').buffer, 'en', 'AD', function (err, result) {
-    assert.notOk(err);
-    assert.end();
-  });
+  assert.end();
 });
 
-test('[localize] success: options may be empty', assert => {
-  localize(mvtFixtures.get('002').buffer, 'en', 'AD', { }, function (err, result) {
-    assert.notOk(err);
-    assert.end();
+test('[localize] params.worldview', (assert) => {
+  localize({
+    buffer: Buffer.from('howdy'),
+    worldview: 1 // not an array
+  }, function (err) {
+    assert.ok(err);
+    assert.equal(err.message, 'params.worldview must be an array', 'expected error message');
   });
-});
 
-test('[localize] success: worldview is not 2 characters long', assert => {
-  assert.plan(6);
-  localize(mvtFixtures.get('002').buffer, 'en', '', function (err) {
+  localize({
+    buffer: Buffer.from('howdy'),
+    worldview: [1, 2, 3] // array with non-strings
+  }, function (err) {
     assert.ok(err);
-    assert.equal(err.message, 'worldview must be a string 2 characters long');
+    assert.equal(err.message, 'params.worldview must be an array of strings', 'expected error message');
   });
 
-  localize(mvtFixtures.get('002').buffer, 'en', 'z', function (err) {
+  localize({
+    buffer: Buffer.from('howdy'),
+    worldview: ['USA'] // array with >2 char strings
+  }, function (err) {
     assert.ok(err);
-    assert.equal(err.message, 'worldview must be a string 2 characters long');
+    assert.equal(err.message, 'params.worldview items must be strings of 2 characters', 'expected error message');
   });
 
-  localize(mvtFixtures.get('002').buffer, 'en', 'abc', function (err) {
+  assert.end();
+});
+
+test('[localize] params.worldview_property', (assert) => {
+  localize({
+    buffer: Buffer.from('howdy'),
+    worldview: ['US'],
+    worldview_property: 1 // not a string
+  }, function (err) {
     assert.ok(err);
-    assert.equal(err.message, 'worldview must be a string 2 characters long');
+    assert.equal(err.message, 'params.worldview_property must be a string', 'expected error message');
   });
+
+  assert.end();
 });
 
-test('[localize] success: worldview is not a string', assert => {
-  localize(mvtFixtures.get('002').buffer, 'en', 100, function (err) {
+test('[localize] params.compress', (assert) => {
```

Comment:
These are all negative tests. Should we have a positive test (maybe just one that has all the values, all set properly)? Or are you planning to rely on other tests (e.g., testing localize functionality) tio do that?

Reply:
I'm not a big fan of separating the validation logic into a separate test file anyways, I think I may move everything into `test/vtcomposite-localize.test.js` eventually. But will keep separate for now and add a positive parameters test for now.